pages/chaoss/visualizations/contrib_importance_pie.py

```diff
@@ -212,51 +233,66 @@ def create_top_k_cntrbs_graph(repolist, action_type, patterns, top_k):
         return dash.no_update, True
 
     # function for all data pre processing
-    df = process_data(df, action_type, patterns, top_k)
+    df = process_data(df, action_type, top_k, patterns, start_date, end_date)
 
     fig = create_figure(df, action_type)
 
     logging.warning(f"{VIZ_ID} - END - {time.perf_counter() - start}")
     return fig, False
 
 
-def process_data(df: pd.DataFrame, action_type, patterns, top_k):
+def process_data(df: pd.DataFrame, action_type, top_k, patterns, start_date, end_date):
+    # convert to datetime objects rather than strings
+    df["created_at"] = pd.to_datetime(df["created_at"], utc=True)
+
+    # order values chronologically by created_at date
+    df = df.sort_values(by='created_at', ascending=True)
+
+    # if the start_date and/or the end date is not specified set them to the beginning and most recent created_at date
+    if start_date is None:
```

Comment:
Can do this more concise: 

`# filter values based on date picker
    if start_date is not None:
        df = df[df.created_at >= start_date]
    if end_date is not None:
        df = df[df.created_at <= end_date]`